public/app/core/services/echo/backends/analytics/GA4Backend.ts

```diff
@@ -0,0 +1,58 @@
+import $ from 'jquery';
+
+import { CurrentUserDTO } from '@grafana/data';
+import { EchoBackend, EchoEventType, PageviewEchoEvent } from '@grafana/runtime';
+
+import { getUserIdentifier } from '../../utils';
+
+declare global {
+  interface Window {
+    dataLayer: unknown[];
+  }
+}
+
+export interface GA4EchoBackendOptions {
+  googleAnalyticsId: string;
+  user?: CurrentUserDTO;
+}
+
+export class GA4EchoBackend implements EchoBackend<PageviewEchoEvent, GA4EchoBackendOptions> {
+  supportedEvents = [EchoEventType.Pageview];
+  trackedUserId: number | null = null;
+
+  constructor(public options: GA4EchoBackendOptions) {
+    const url = `https://www.googletagmanager.com/gtag/js?id=${options.googleAnalyticsId}`;
+
+    $.ajax({
+      url,
+      dataType: 'script',
+      cache: true,
+    });
```

Comment:
Maybe we could replace $ with `fetch`?

Reply:
Yup, would love to do this for both the backends!